.github/workflows/test-suite.yml

```diff
@@ -39,48 +29,70 @@ jobs:
       id: changed-files
       uses: tj-actions/changed-files@v45
       with:
-        files: |
-          archive/**
-          test/**
-          .glotter.yml
-          .github/workflows/test-suite.yml
-          pyproject.toml
-          poetry.lock
-          !**/README.md
-
-    - name: Show changed code files
+        since_last_remote_commit: true
+        files_ignore: |
+          **/*.md
+
+    - name: List all changed files markdown files
+      if: steps.changed-files.outputs.any_changed == 'true'
       env:
         ALL_CHANGED_FILES: ${{ steps.changed-files.outputs.all_changed_files }}
       run: |
-        echo "Changed code files:"
         for file in ${ALL_CHANGED_FILES}; do
-          echo "- $file"
+          echo "$file was changed"
         done
 
+  # We need this job to test all code in the codebase
+  testing:
+    name: "Test Suite"
+    needs: changed-files
+    if: needs.changed-files.outputs.status == 'true'
+
+    strategy:
+      fail-fast: true
+      matrix:
+        python-version: ["3.11"]
+        poetry-version: ["1.8.4"]
+        os: [ubuntu-latest]
+        num_batches: [6]
+
+    runs-on: ${{ matrix.os }}
+    steps:
+    - uses: actions/checkout@v4
+      with:
+        fetch-depth: ${{ github.event_name == 'pull_request' && 2 || 0 }}
```

Comment:
I was thinking about this, and I'm wondering if this is right. It seems like for a pull request, you'd want the full history of that pull request (`0`), and for a push, you'd only want the previous commit. Not sure what number should be used for that, but I suspect it might look like this:
```yaml
fetch-depth: ${{ github.event_name == 'push' && 2 || 0 }}
```
I originally got this line from a StackOverflow article (don't remember the source though)